ui/App/SettingsScreen.svelte

```diff
@@ -95,6 +118,45 @@
   ];
 
   const session = Session.unsealed();
+
+  let colorHex: string = "";
+
+  const hexToRgb = (hex: string) => {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return result
+      ? {
+          r: parseInt(result[1], 16),
+          g: parseInt(result[2], 16),
+          b: parseInt(result[3], 16),
+        }
+      : {
+          r: 21,
+          g: 21,
+          b: 21,
+        };
+  };
+
+  $: if ($primaryColorStore === "hex") {
+    const colorRgb = hexToRgb(colorHex);
+
+    const element = document.createElement("style");
+    // Append style element to head
+    document.head.appendChild(element);
+    // Reference to the stylesheet
+    // eslint-disable-next-line
+    const sheet = element.sheet!;
+    let styles = '[data-primarycolor="hex"] {';
+    styles += `--color-primary: rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},1`;
+    styles += `--color-primary: rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.17`;
+    styles += `--color-primary: rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.37`;
+    styles += `--color-primary: rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.87`;
+    styles += "}";
+
+    // Add the first CSS rule to the stylesheet
+    sheet.insertRule(styles, 0);
+  }
+
+  $: console.log($primaryColorStore);
```

Comment:
@geigerzaehler @rudolfs trying it this approach and it's not working. I don't know how I can check if the new stylesheet even exists. Also I don't want to pull in master cause that's gonna break everything.